### PR TITLE
Use the GCE_METADATA_HOST environment variable when loading metadata

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: Google.Api.Gax.Tests/EnvironmentVariableRestorer.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+using System;
+
+namespace Google.Api.Gax.Tests
+{
+    /// <summary>
+    /// Helper to restore the value of an environment variable when it's disposed.
+    /// </summary>
+    internal sealed class EnvironmentVariableRestorer : IDisposable
```
Comment (amanda-tarafa) [inline Google.Api.Gax.Tests/EnvironmentVariableRestorer.cs:15]: This is cool!!

File: Google.Api.Gax/Platform.cs [hunk 2]
```diff
@@ -98,6 +96,30 @@ private static async Task<string> LoadMetadataAsync()
             }
         }
 
+        /// <summary>
+        /// Determine the metadata host to use. In order of priority, this will use:
+        /// - GCE_METADATA_HOST environment variable, if set and non-empty
+        /// - METADATA_EMULATOR_HOST environment variable, if set and non-empty.
```
Comment (amanda-tarafa) [inline Google.Api.Gax/Platform.cs:102]: I would have thought the emulator comes first.

File: Google.Api.Gax/Platform.cs [hunk 3]
```diff
@@ -98,6 +96,30 @@ private static async Task<string> LoadMetadataAsync()
...
+        ///   This is the undocumented but the de-facto mechanism for doing using an emulator.
```
Comment (amanda-tarafa) [inline Google.Api.Gax/Platform.cs:103]:
```suggestion
        ///   This is the undocumented but the de-facto mechanism for using an emulator.
```

**Review by amanda-tarafa — APPROVED** [review]
LGTM

File: Google.Api.Gax/Platform.cs (hunk 3 above)
Comment (jskeet) [reply Google.Api.Gax/Platform.cs:103]: Done, thanks.

File: Google.Api.Gax/Platform.cs (hunk 2 above)
Comment (jskeet) [reply Google.Api.Gax/Platform.cs:102]: Confirmed that GCE_METADATA_HOST should take priority. Merging.

**Pull request merged**